### PR TITLE
ui: Redesign user interface

### PR DESCRIPTION
**Pull request by mborgerson**
Introduces a new user interface that looks much nicer, is easier to
navigate with controllers, provides more context to users, and is
scalable. Some additional features are included.

* Adds 'popup menu' with actions that can be used easily from controller
* Adds 'main menu', unifying other configuration dialogs
* Adds port-forwarding user interface
* Adds screenshot feature
* Adds volume control feature
* Adds gamepad auto-bind option
* Adds vsync configuration option
* Adds auto UI scaling
* Adds preferred window size selection
* Adds AV pack selection
* Exposes some existing config items in GUI

### CODE REVIEW

**Comment by GXTX** [comment]
Size saving is broken, and AV pack should probably be allowed to be swapped while running, Xbox has a SMC interrupt thing to notify the OS and it's on the OS to decide what it wants to do. Looks good overall!

**Comment by mborgerson** [comment]
> Size saving is broken, and AV pack should probably be allowed to be swapped while running, Xbox has a SMC interrupt thing to notify the OS and it's on the OS to decide what it wants to do. Looks good overall!

@GXTX Agree! I'll accept a PR for this. I added smc interrupts a while back so I imagine it's probably an easy add.

File: config_spec.yml [hunk 1]
```diff
@@ -16,26 +18,57 @@ input:
...
     surface_scale:
       type: integer
       default: 1
   window:
-    last_width: integer
-    last_height: integer
+    fullscreen_on_startup: bool
+    startup_size:
+      type: enum
+      values: [last_used, 640x480, 1280x720, 1280x800, 1280x960, 1920x1080, 2560x1440, 2560x1600, 2560x1920, 3840x2560]
```
Comment (MasonT8198) [inline config_spec.yml]: I'd consider adding 3840x2160 to this, 2560 is an odd resolution

Comment (mborgerson) [reply config_spec.yml]: You are right! That should have been 2160

Comment (MasonT8198) [reply config_spec.yml]: Alright, thanks for the fix!

Comment (MasonT8198) [reply config_spec.yml]: I'd consider adding 1366x768 and 1600x900 as they are still pretty common laptop resolutions, even on some older machines with dgpus in them

**Pull request merged**